### PR TITLE
image optimization

### PR DESCRIPTION
**Pull request by joekarasek**
New all in one function works, but I'm not seeing a significant boost in compile time :(

### CODE REVIEW

File: packages/components/bolt-image/src/image.twig [hunk 1]
```diff
@@ -160,7 +131,7 @@
 {% endset %}
 
 {# Super helpful info if you uncomment #}
-{#{{ console_log(_context) }}#}
+{{ console_log(_context) }}
```
Comment (sghoweri) [inline packages/components/bolt-image/src/image.twig]: Is this necessary? Aren't we already outputting context data everywhere? Or is that just on the static site vs Pattern Lab...?

**Comment by sghoweri** [comment]
Hold off on merging this just yet. Seeing some really strange performance numbers I think we need to dig into (600KB+ sized images being downloaded on mobile):

https://www.webpagetest.org/result/180308_VF_08adfcb7c99a6f2bdda4544d16455fe2/1/details/#waterfall_view_step1

File: packages/build-tools/tasks/image-tasks.js [hunk 2]
```diff
@@ -117,8 +117,8 @@ async function processImage(file, set) {
     const sets = resizedImagePaths.filter(resizedImagePath => resizedImagePath);
     const imageMeta = {
       original: file,
-      width,
-      height,
+      // width,
```
Comment (sghoweri) [inline packages/build-tools/tasks/image-tasks.js]: @joekarasek out of curiosity, why are `height` and `width` commented out here? Should these be removed?

File: packages/components/bolt-image/image.schema.yml [hunk 3]
```diff
@@ -18,10 +18,24 @@ properties:
...
-  zoom:
+  srcset:
+    type: string
+    description: A comma seperated string of image urls and page widths, used for optimizing image loading performance.
+    default: Generated during build.
+  sizes:
+    type: string
+    description: A list of one or more strings separated by commas indicating a set of source sizes. Each source size consists of a media condition (omitted for the last item), and a source size value.
+    default: auto
+  useAspectRatio:
```
Comment (sghoweri) [inline packages/components/bolt-image/image.schema.yml]: +1

File: packages/components/bolt-image/src/image.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+const readFile = promisify(fs.readFile);
+const path = require('path');
+
+describe('image file output', () => {
+
+  test('dummy test', () => {
+    expect(true).toEqual(true);
+  });
+
+  // @todo Build out snapshot testing with Jest for images
```
Comment (sghoweri) [inline packages/components/bolt-image/src/image.test.js]: @joekarasek we should think about adding some follow-up tasks so this todo can be checked off ;-)

File: packages/core-php/tests/ImagesTest.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+   * @param $precision float acceptable percent difference between the two numbers to return true.
+   * @return bool
+   */
+  private static function compareColors($expectedColor, $actualColor, $precision) {
+    $expectedColorValue = hexdec($expectedColor);
+    $actualColorValue = hexdec($actualColor);
+
+    return abs(($expectedColorValue-$actualColorValue)/$expectedColorValue) < $precision;
+  }
+}
```
Comment (sghoweri) [inline packages/core-php/tests/ImagesTest.php]: <3

File: packages/components/bolt-image/src/image.twig [hunk 6]
```diff
@@ -1,109 +1,19 @@
...
-#}
-
-{#{% set srcset = srcset ? srcset : "" %}#}
-
-{# Grabbing the data in the bolt manifest for this particular image #}
-{% set thisImageData = bolt.data.images[src] %}
-
-{# @todo DRY this #}
-{% set imageSizes = [
-  50, 100, 200, 320, 480, 640, 800, 1024, 1366, 1536, 1920, 2560, 2880
```
Comment (sghoweri) [inline packages/components/bolt-image/src/image.twig]: @joekarasek question: with these image size values getting removed here - where is the array of the different image size cut options living?

File: packages/build-tools/tasks/image-tasks.js (hunk 2 above)
Comment (joekarasek) [reply packages/build-tools/tasks/image-tasks.js]: @sghoweri, @EvanLovely and I had some back and forth on this. We decided to cut down on user confusion by allowing only one source of height and width, and we weren't providing all the data in the global bolt.data (all or nothing). Was basically redundant with reading the image with twig and php

**Pull request merged**